radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/meter/TxnSizeFeeMeter.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ */
+
+package com.radixdlt.constraintmachine.meter;
+
+import com.radixdlt.application.tokens.state.TokensInAccount;
+import com.radixdlt.constraintmachine.ExecutionContext;
+import com.radixdlt.constraintmachine.ProcedureKey;
+import com.radixdlt.constraintmachine.REOp;
+import com.radixdlt.constraintmachine.SubstateWithArg;
+import com.radixdlt.utils.UInt256;
+
+public class TxnSizeFeeMeter implements Meter {
+	private final UInt256 feePerByte;
+
+	private TxnSizeFeeMeter(UInt256 feePerByte) {
+		this.feePerByte = feePerByte;
+	}
+
+	public static TxnSizeFeeMeter create(UInt256 feePerByte) {
+		return new TxnSizeFeeMeter(feePerByte);
+	}
+
+	@Override
+	public void onUserProcedure(ProcedureKey procedureKey, Object param, ExecutionContext context) throws Exception {
+		context.chargeOneTimeTransactionFee(txn -> UInt256.from(txn.getPayload().length).multiply(feePerByte));
```

Comment:
Is it impossible that the initial loan does not cover the transaction cost?

Reply:
Only if Its a large transaction (thats one of the things the loan is supposed to protect against.